chainercv/datasets/coco/coco_instance_segmentation_dataset.py

```diff
@@ -0,0 +1,183 @@
+    5000 images in the validation images. The remaining validation
+    images are called 'minvalminus'. Concrete list of image ids and
+    annotations for these splits are found `here`_.
+
+    .. _`here`: https://github.com/rbgirshick/py-faster-rcnn/tree/master/data
+
+    .. [#] Tsung-Yi Lin, Michael Maire, Serge Belongie, Lubomir Bourdev, \
+        Ross Girshick, James Hays, Pietro Perona, Deva Ramanan, \
+        C. Lawrence Zitnick, Piotr Dollar.
+        `Microsoft COCO: Common Objects in Context \
+        <https://arxiv.org/abs/1405.0312>`_. arXiv 2014.
+
+    Args:
+        data_dir (string): Path to the root of the training data. If this is
+            :obj:`auto`, this class will automatically download data for you
+            under :obj:`$CHAINER_DATASET_ROOT/pfnet/chainercv/coco`.
+        split ({'train', 'val', 'minival', 'valminusminival'}): Select
+            a split of the dataset.
+        use_crowded (bool): If true, use bounding boxes that are labeled as
+            crowded in the original annotation.
+        return_crowded (bool): If true, this dataset returns a boolean array
+            that indicates whether bounding boxes are labeled as crowded
+            or not. The default value is :obj:`False`.
+        return_area (bool): If true, this dataset returns areas of masks
+            around objects.
+
+    """
+
+    def __init__(
+            self, data_dir='auto', split='train',
+            use_crowded=False, return_crowded=False,
+            return_area=False
+    ):
+        if not _availabel:
+            raise ValueError(
+                'Please install pycocotools \n'
+                'pip install -e \'git+https://github.com/pdollar/coco.git'
+                '#egg=pycocotools&subdirectory=PythonAPI\'')
+
+        super(COCOInstanceSegmentationDataset, self).__init__()
+        self.use_crowded = use_crowded
+
+        if split in ['val', 'minival', 'valminusminival']:
+            img_split = 'val'
+        else:
+            img_split = 'train'
+        if data_dir == 'auto':
+            data_dir = get_coco(split, img_split)
+
+        self.img_root = os.path.join(
+            data_dir, 'images', '{}2014'.format(img_split))
+        anno_path = os.path.join(
+            data_dir, 'annotations', 'instances_{}2014.json'.format(split))
+
+        self.data_dir = data_dir
+        annos = json.load(open(anno_path, 'r'))
+
+        self.id_to_prop = {}
+        for prop in annos['images']:
+            self.id_to_prop[prop['id']] = prop
+        self.ids = sorted(list(self.id_to_prop.keys()))
+
+        self.cat_ids = [cat['id'] for cat in annos['categories']]
+
+        self.id_to_anno = defaultdict(list)
+        for anno in annos['annotations']:
+            self.id_to_anno[anno['image_id']].append(anno)
+
+        self.add_getter('img', self._get_image)
+        self.add_getter(['mask', 'label', 'area', 'crowded'],
+                        self._get_annotations)
```

Comment:
If we can retrieve 'label', 'area', 'crowded' without loading 'mask', it is better to separate getter function.

Reply:
Can you tell me why?
`self.use_crowded` is set as `True`, `crowded` is loaded for every annotation.

Reply:
I mean making `get_mask` and `get_label_area_crowded`.

Reply:
I know `mask` requires `crowded` in some cases. However, `label` or `area` can be retrieved without loading `mask`.

Reply:
So, in `get_mask`, `crowded` is loaded, but not returned?

Reply:
@Hakuyume can you give me more advice? spliting into two function is efficient for data load?

Reply:
He is saying that if `_get_annotation` is split into two, we can access `labels` without `masks` using `dataset.slice[:, 'slice']`.
We are assuming that loading `masks` takes substantially longer time than loading `labels`.

Reply:
> So, in _get_mask, crowded is loaded, but not returned?

Yes, if use_crowded=False, `_get_mask` calls `_get_label_area_crowded` internally to get `crowded`. If use_crowded=True, `_get_mask` doesn't need to call `_get_label_area_crowded`.